### ISSUE TITLE
Monitor will not load after update

### ISSUE THREAD

**Issue by RoyRomeo** [jgyates/genmon#266]
Use the template below if you have an issue or want to report a bug. If you have a question or a feature request you can ignore the questions below.

***NOTE:*** If you are having issues with your serial connection, please read this page before posting:

https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting

If you are having other issues, please see the following page:

https://github.com/jgyates/genmon/wiki/3.5---General-Troubleshooting

If you need to send you logs and registers to the developer, if you email is setup and working properly you can click send your logs on the About page in the web interface.

### Expected Behavior

Generator monitor to load

### Actual Behavior

After most recent update, screen shows "Loading Generator Monitor" but does not load.

### Steps to Reproduce (including precondition)

{Please write here}

### Screenshot or Pictures relating to the problem (if possible)

{Please write here}

### Your Environment

- Generator Model: {Please write here}
- Generator Registers: Submit via the About page
- Genmon Version: {Please write here}

**Comment by jgyates** [comment]
Are you using V1.13.08 or V1.13.09?  V1.13.09 was posted about an hour or two ago.

Try this to get you back to a known state. Log into the pi and get into the genmon directory:

     ./genmonmaint.sh -u
     cp ./conf/genloader.conf /etc/genmon 
     ./startgenmon.sh restart

   
Let me know how that goes.

**Comment by SethL** [comment]
I'm having the same exact issue.  I followed the steps you just suggested and no go.  It still just hangs on loading in the webui.  After running the commands you suggested, my device lost ssh and vnc for a while.  then i get an email saying genmon is not communicating with the controller.  Now I have VNC and ssh back but the webui is still stuck on loading

**Comment by RoyRomeo** [comment]
I'm using the one that was available yesterday.  Do you log in to the Pi via Putty?  If so, what is the command line to access the Pi?


-----Original Message-----

**Comment by SethL** [comment]
![Capture](https://user-images.githubusercontent.com/2295127/57587390-3ef97200-74d2-11e9-8852-a310e2a37ca3.PNG)
 
Could it be rebuilding the db or something?  the CPU is pegged at 100% and genmon.py is consuming it via multiple threads. see attached

**Comment by jgyates** [comment]
Try a reboot, then see if that makes a difference. I am not sure why it is locking, but the update from today is needed as I discovered a problem with the file copy. 

If you have a backup you can unzip the files and copy them into the /etc/genmon directory, then reboot. The base issue that appears to be happening is that the config files are being corrupted during the copy. the new version of the software moves all the config files from /etc/ to /etc/genmond to better support multiple instances. but V1.13.08 had a problem that could occur if the system did not copy the files fast enough, they could be corrupted.

Also, you can alway delete the files in the /etc/genmon/ directory and the software should re-write them with a fresh copy. This is the equivalent of resetting all settings to the original.

**Comment by jgyates** [comment]
one thing I am trying to determine is this: Is there a problem doing an upgrade with V1.13.09. Know there is a problem that can occur if you upgrade to V1.13.08, but I think I fixed that with V1.13.09. If anyone has tried this (besides me) and can validate that would be great.

**Comment by SethL** [comment]
Im not sure what version I updated to.. I left the UI open for a day or so and then clicked update.  so it may have updated to 1.13.08.  I did create a backup before I did it so once I copied the files over and restarted it worked like a charm :-)

Thanks!

**Comment by swvalenti194** [comment]
1.13.08 updated but genmom wouldn't start. Had to reboot pi and it came up. Updated to .09 with zero issues came right up. Only thing I noticed is the notification via MMS on ATT took days or hours to notify on change etc...

**Comment by RoyRomeo** [comment]
I tried a reboot of the pi, and can no longer connect.

**Comment by jgyates** [comment]
Did you have your pi setup for a static or dynamic IP address? If it is dynamic then the reboot may have caused your router to issue a new IP address. If this is the case then logging into your router may give you some insight as to the new IP address. If it is setup for a static IP and you can not connect then your best best is to connect a keyboard and monitor and try to determine what is not working.

**Comment by RoyRomeo** [comment]
When I type in "wifisetup.pi:1880/GenPi", I get a "Welcome to the Node-RED Dashboard, Please add some UI nodes to your flow and redeploy.

**Comment by SethL** [comment]
You should connect to your pi directly with kbd and monitor if you cannot
connect remotely.

**Comment by liltux** [comment]
@RoyRomeo did you reset your network setup?  The wifisetuo.pi:1880/GenPi only works if your computer is connected to the genpi network when you reset your network settings and restart the AP.  Else you would type in the IP address of the pi and :1880/GenPi to access the node red UI.  You may have to use your router or a free app like FING to scan your network and identify your Pi's IP address.

**Comment by jgyates** [comment]
@RoyRomeo 

I believe you have the wifi config from @liltux. He may contact you about this if he has any suggestions regarding the Node-Red config.

If you are familure with ping you should be able to open a command prompt or terminal session and type:

    ping wifisetup.pi

If this is successful then you know that your IP is still on your network.

From there you should be able to log into your pi using the ssh program. the Node-Red configuration is something that @liltux has setup on the the pre-configured pi available on e-bay.

**Comment by liltux** [comment]
@RoyRomeo you can email me directly at liltux @ liltux dot com.

**Comment by jcherney** [comment]
Same problem here. Loaded version 1.13.09 and it just spins waiting for the page to load. Tried using 
 ./genmonmaint.sh -u
 cp ./conf/genloader.conf /etc/genmon 
 ./startgenmon.sh restart

But after this command:    cp ./conf/genloader.conf /etc/genmon   I get a    cp:missing destination file operand after './conf/genloader.conf/etc/genmon'

Thoughts?

**Comment by jcherney** [comment]
Actually, ./genmonmaint.sh -u errors out also.  no such file or directory

**Comment by jcherney** [comment]
Ok, I wasnt in the genmon directory, sorry.  The Script uploaded fine, but the next command still gets an error or command not found.

**Comment by jcherney** [comment]
![image](https://user-images.githubusercontent.com/37984103/57973864-4f4f9800-797d-11e9-9ac1-ae2cfaddbed0.png)

**Comment by liltux** [comment]
@jcherney what version were you updating from?  Might try:
sudo bash genmonmaint -i 
This will reinstall everything.  Also if doing this backup your current conf files so they don't get overwritten.

**Comment by jcherney** [comment]
@liltux  I think it was 1.13.07   Your command suggestion gives no such file or directory.

**Comment by jcherney** [comment]
I got the genmontmaint.sh -u to work- see above screenshot.  It's the next command that doesnt

**Comment by liltux** [comment]
Oops, 
sudo bash /home/pi/genmon/genmonmaint.sh -i

**Comment by liltux** [comment]
@jcherney the script is trying to copy or verify files and either files or directory is missing.

**Comment by jcherney** [comment]
Can I back up the config files from the ssh session before I issue that command?

PS - I can do a hard restart outside. Would that eliminate any of this ?

**Comment by liltux** [comment]
The resetting outside would more than likely not affect.  Look for Gen*.conf files into the /etc directory.  You can then cp them to ~/ directory (home folder shorthand).
Easy command
ls /etc | grep gen*
See what is listed, bet most are the files that you need.

**Comment by jcherney** [comment]
![image](https://user-images.githubusercontent.com/37984103/57974037-f5040680-797f-11e9-8158-fc42ed9e1756.png)

**Comment by liltux** [comment]
@jcherney

ls(space)/etc(space)|(space)grep(space)gen*

ls /etc | grep gen*

I see, I left off a colon.  It's LS not is, sorry.

**Comment by jcherney** [comment]
Sorry to be a pain in the ass, but: 
![image](https://user-images.githubusercontent.com/37984103/57974111-f681fe80-7980-11e9-9718-bf5d5e1bd0d5.png)

**Comment by liltux** [comment]
Leave off the wild card, not thinking straight.

**Comment by liltux** [comment]
Also will need mymail.conf

**Comment by jcherney** [comment]
![image](https://user-images.githubusercontent.com/37984103/57974183-5fb64180-7982-11e9-9728-7b4b2b876864.png)

**Comment by jcherney** [comment]
![image](https://user-images.githubusercontent.com/37984103/57974256-45309800-7983-11e9-9193-897d8034eba7.png)

**Comment by liltux** [comment]
Now you need to issue the copy command for those files.
Example:
cp /etc/mymail.conf ~/
Explained:
cp space orig file and location space new location.  Copy my mail.conf from etc to ~/ folder which is short for current user home directory

**Comment by jcherney** [comment]
Ok, done. mymail.conf doesnt' seem to exist there though.
![image](https://user-images.githubusercontent.com/37984103/57974450-40211800-7986-11e9-84b5-6bcac53f148a.png)

**Comment by liltux** [comment]
It does, just do the run the cp command for mymail.conf just like you did the rest

**Comment by jcherney** [comment]
Done.

**Comment by jcherney** [comment]
Now run this?  sudo bash /home/pi/genmon/genmonmaint.sh -i

**Comment by liltux** [comment]
Yes

**Comment by jcherney** [comment]
Do I want to copy the configuration files to /etc/genmon?

**Comment by jcherney** [comment]
![image](https://user-images.githubusercontent.com/37984103/57974627-7318db00-7989-11e9-96e9-ecf79840f35b.png)

**Comment by liltux** [comment]
Answer yes to everything.  Then you will copy your back up files to /etc/genmon yes.

**Comment by jcherney** [comment]
Ok, script is finished. what directory do i need to be in to do the copy back?

**Comment by liltux** [comment]
In pi home dir.  Type:

cd ~/

**Comment by jcherney** [comment]
so  cp gengpioin.conf /etc/genmon ?

**Comment by liltux** [comment]
That would be correct syntax

**Comment by jcherney** [comment]
![image](https://user-images.githubusercontent.com/37984103/57974803-fdfad500-798b-11e9-8980-1f0bf3496ab3.png)

**Comment by liltux** [comment]
Sorry again, you will have to perform the action as sudo, so:
sudo cp ....

**Comment by jcherney** [comment]
No worries. Done!

**Comment by jcherney** [comment]
now this?  ./startgenmon.sh restart

**Comment by liltux** [comment]
Best just to type:
sudo reboot
This will allow the pi to restart and run the scripts automatic

**Comment by jcherney** [comment]
Awesome! It's back! Once again, thank you SO much!   What do you think is causing this?

**Comment by liltux** [comment]
What I have noticed is that once performing the update on the webUI, on my
images(before latest) I have also been having to ssh in or using my NodeRed
webUI and reboot.  Works every time.  But when I am unsure what you may
have done, it's always best to go through the install again.  Makes sure
all needed files are where they should be.  The latest has update on GenMon
has worked for me, again I just have to issue a reboot to the pi and
everything comes back up.

**Comment by jcherney** [comment]
What is NodeRed?

**Comment by liltux** [comment]
Nice way to program IOT and other things using node and js.  NodeRed.org

**Comment by jcherney** [comment]
Thanks.

**Comment by jgyates** [comment]
Sorry for the late response, I was helping my son move today so I was away from the internet until now. Sounds like @liltux got you back up and running.

@jcherney : One question, are you running on a pi3 or pi zero? I have a delay to let the file system catch up in the upgrade code that runs only once when upgrading to 1.13.09 (it does not need to run in future versions). I am wondering if the the pi zero may need a longer delay. another thing that could cause this is an unusually slow SD card.

**Comment by jcherney** [comment]
He did. He’s a huge help

I’m running on a Pi3

**Comment by jcherney** [comment]
Hope the move went well!

**Comment by liltux** [comment]
@jgyates how much of a delay do you have in the update?  I am wondering if I'm getting in too much of a hurry?  All of my images are now up with the latest revision.  I did work on 2 today that had older revisions and after update, once I saw the spinning wheel, I just ran sudo reboot and once restarted everything was good.   No to say there was a problem before, I was just impatient.

**Comment by gmqest** [comment]
Just upgraded pi zero from V1.13.07  one hour ago and encountering the same issue: hangs on loading the webui. 

After a reboot, the pi zero seems to be doing something - CPU usage at ~30%, load average ~0.43. Python using all resources and no other services running on the pi. Will leave it alone for a while to see if it starts.

**Comment by jgyates** [comment]
@gmqest 

Around 30% is normal for a pi unless you have the "Optimize for slower CPUs" setting enabled. the core issue appears to be slow SD card access. During the upgrade all genmon related conf files are copied from the /etc/ directory to the /etc/genmon directory and the power log and outage log are copied from the /home/pi/genmon directory to the /etc/genmon/ directory. V1.13.09 waits one second before attempting to read / modify the newly copied files. If the SD card has not finished the copy then the files may become corrupt. To fix this issue you can either copy your backup files to the /etc/genmon/ directory or start clean and copy the contents of /home/pi/genmon/conf/ to /etc/genmon/. I don't know if this is specific to the pi zero being a much slower CPU or if it specific to SD cards that have a slower response time. Ideally you should be using a class 10 or better SD card. I can make the delay longer to try to compensate, but slower SD cards can potentially run into other issues with Linux in general.

Let me know if you have any questions regarding how to resolve this problem.

**Comment by jgyates** [comment]
I have checked in V1.13.11 that hopefully will fix these issues for thoes who have not upgraded from 1.13.07 or earlier.

**Comment by gmqest** [comment]
Reinstalling with this version (V1.13.11), then overwriting conf files from a recent backup worked for me. Other suggestions above did not.

**Comment by jcherney** [comment]
FYI, my SD card is a Sandisk Ultra 10.

**Comment by jgyates** [comment]
I am closing this issue as I believe it has been fixed and I have had no other reports of problems for 2 weeks. If anyone still has issues feel free to comment in this thread.

**Comment by Machinedoc** [comment]
Just upgraded from V1.12.31 to V1.13.12. Got message that "Genmon is loading..." but Genmon never came up.  Did a cold start of Pi and Genmon came up OK.

**Comment by jgyates** [comment]
Hi @Machinedoc 

Thanks for the info. This should like a separate issue, but if you want to send your log files from the about page (outbound email must be working for this to work) I could take a look at your logs and see if anything out of the ordinary shows up.

**Comment by Machinedoc** [comment]
I think outbound email is working; I just sent logs and registers - please
let me know if you received them.

You can also send email to michael@mirsky.net

Not for nothing - I think Genmon is a real winner!!!  Why isn't Generac
banging on your door with a big, fat check?  Clearly, this is not for
everyone, but their Mobile link just looks like a loser.  I believe they
started incorporating wifi into some of their residential machines in May
or June of last year.  (My machine was manufactured in April 2018 - it was
installed in July , but not up and running until Decemeber because my
natural gas supplier made life miserable for us.  I finally got to
installing the cable from the machine to the Pi 4 days ago.  This is just
fabulous...Thank You!

Mike

**Comment by jgyates** [comment]
Hi Mike,

Thanks for the kind words. Your logs look good except for an error that is likely caused by a dropped serial packet. My guess is that on the restart where you were having problems the pi missed reading the register that IDs the controller. this caused the software to think your controller was a Nexus and as a result some data relating to the maintenance timers were interpreted incorrectly (Nexus and Evolution have different format for some of the registers). I will look over the related code and see if I can make this a bit more robust. Let me know if you see the problem again.

Thanks
Jason
